### PR TITLE
 Fix windows numbering issue due to changing port numbers

### PR DESCRIPTION
**Pull request by cpyarger**
in the windows midi library, enumeration of port names was based on port numbers, which is widely variable based on the number of devices plugged in, when the os sees plugged in devices, and a few other reasons.  This checks if a device exists, if it does, it increments the enumeration based on the number of devices with that name.

### CODE REVIEW

File: rtmidi17/detail/winmm.hpp [hunk 1]
```diff
@@ -363,12 +366,25 @@ class midi_in_winmm final : public midi_in_default<midi_in_winmm>
...
+      int x = 1;
+      for (int i = 0; i < portNumber; i++)
+      {
+        MIDIINCAPS deviceCaps2;
+        midiInGetDevCaps(i, &deviceCaps2, sizeof(MIDIINCAPS));
+        auto stringName2 = ConvertToUTF8(deviceCaps2.szPname);
+        if (stringName == stringName2)
+        {
+          x++;
+	}
```
Comment (jcelerier) [inline rtmidi17/detail/winmm.hpp:378]: can you run clang-format ? had a double take when I saw that code :-)

File: rtmidi17/detail/winmm.hpp [hunk 2]
```diff
@@ -98,6 +95,10 @@ class observer_winmm final : public observer_api
     inline static const bool OUTPUT = false;
 
 public:
+    using PortList = std::vector<std::string>;
+
+    PortList inputPortList;
```
Comment (jcelerier) [inline rtmidi17/detail/winmm.hpp:100]: the typedef can stay public, but I think that the member variables should stay private

**Comment by jcelerier** [comment]
Thanks for the pull request. I gotta check that it doesn't break things as I had some projects that relied on the existing status (maybe based on pure luck ? but it seemed to work so far) so need to at least see that things aren't gonna explode :-) but it's very appreciated,  should merge it shortly

**Comment by cpyarger** [comment]
I haven't quite got clang-format figured out on windows yet.
and yes, I originally was going to do a search of the portslists for
existing members with the same name, I forgot to revert that change.
Warmest Regards,


Christopher P. Yarger

Phone: 802-505-7574
Skype: cpyarger

/************************************
Based upon email reliability concerns,
please send an acknowledgment in response to this note.

**Comment by jcelerier** [comment]
> I haven't quite got clang-format figured out on windows yet.

okay, don't worry about it then I'll do it (it can be a bit of a pain to install there indeed if you don't have e.g. MSYS)

**Comment by jcelerier** [comment]
things seem good - I'll merge and apply clang-format. thanks :-)

**Pull request merged**